tests/test_connection.py

```diff
@@ -415,23 +431,40 @@ async def handler(request):
         assert header_key == b'x-test-header'
         assert header_value == b'My test header'
 
+def _trio_default_loose() -> bool:
+    assert re.match(r'^0\.\d\d\.', trio.__version__), "unexpected trio versioning scheme"
+    return int(trio.__version__[2:4]) < 25
 
 @fail_after(1)
-async def test_handshake_exception_before_accept():
+async def test_handshake_exception_before_accept() -> None:
     ''' In #107, a request handler that throws an exception before finishing the
     handshake causes the task to hang. The proper behavior is to raise an
     exception to the nursery as soon as possible. '''
     async def handler(request):
         raise ValueError()
 
-    with pytest.raises(ValueError):
+    with pytest.raises((BaseExceptionGroup, ValueError)) as exc:  # pylint: disable=possibly-used-before-assignment
```

Comment:
for cases like this, I would rather see an explicit `assert exc` in the code over a pylint annotation

Reply:
it's `BaseExceptionGroup` that pylint fails to realize will always be available. But can switch to an `assert`

Reply:
...except mypy correctly picks up that it will always be truthy, and raises a warning if doing `assert BaseExceptionGroup` >.>